app/helpers/application_helper.rb

```diff
@@ -164,9 +164,31 @@ def show_institution?(action, resource)
     ComputedPolicy.condition_resources_for(action, resource, current_user)[:institution].count > 1
   end
 
-  def validation_errors(model)
-    if model.errors.present?
-      render partial: "shared/validation_errors", locals: { model: model }
+  def validation_errors(model, attr_name = nil)
+    messages =
+      if attr_name
+        model.errors.full_messages_for(attr_name.to_sym)
+      else
+        model.errors.full_messages
+      end
+
+    unless messages.try(&:empty?)
+      render partial: "shared/validation_errors", locals: { messages: messages }
+    end
+  end
+
+  def field_with_error(model, attr_name, **options)
+    if messages = model.errors.get(attr_name.to_sym)
+      options[:class] = [*options[:class], "field_with_errors"]
+
+      content_tag :div, **options do
+        yield
+        concat content_tag(:div, messages.first, class: "error-message")
+      end
+    else
+      content_tag :div, **options do
+        yield
+      end
```

Comment:
Is this integral to this PR? I'd rather leave this out and introduce improvements to the form builder separately.

Reply:
Let's add a NOTE to rewrite as a proper form builder, and call it a day :bow: ?

Reply:
Okay fine... I just don't like that we keep adding little bits and pieces everywhere that we need to clean up again afterwards.

Reply:
I only noticed `#field_errors` that kinda replicates `#field_with_errors` but it was used only once, so I merged the two methods together, so they match Joni's design.

I added a TODO note that `#field_with_error` and `#validation_errors` should be extracted into a custom FormBuilder!